File: src/main/java/skywolf46/extrautility/util/MaterialUtil.java
package skywolf46.extrautility.util;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import org.bukkit.Material;

public class MaterialUtil {


    public static ItemTypeData parseStringName(String type) {
        short durability = 0;
        if (type.contains(":")) {
            String[] split = type.split(":");
            type = split[0];
            durability = Short.parseShort(split[1]);
        }
        Material data = Material.matchMaterial(type);
        if (data == null)
            return null;
        return new ItemTypeData(data, durability);
    }

    @AllArgsConstructor
    @EqualsAndHashCode
    public static final class ItemTypeData {
        @Getter
        @EqualsAndHashCode.Include
        private Material type;
        @Getter
        @EqualsAndHashCode.Include
        private short durability = 0;

    }
}

File: src/main/kotlin/skywolf46/extrautility/util/MaterialUtil.kt
package skywolf46.extrautility.util

import org.bukkit.Material
import skywolf46.extrautility.util.MaterialUtil.ItemTypeData




class MaterialUtil {
    fun parseStringName(type: String): ItemTypeData? {
        var type = type
        var durability: Short = 0
        if (type.contains(":")) {
            val split = type.split(":").toTypedArray()
            type = split[0]
            durability = split[1].toShort()
        }
        val data = Material.matchMaterial(type) ?: return null
        return ItemTypeData(data, durability)
    }

    class ItemTypeData(val type: Material, val durability: Short) {
        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (javaClass != other?.javaClass) return false

            other as ItemTypeData

            if (type != other.type) return false
            if (durability != other.durability) return false

            return true
        }

        override fun hashCode(): Int {
            var result = type.hashCode()
            result = 31 * result + durability
            return result
        }
    }
}